pytorch_lightning/trainer/training_io.py

```diff
@@ -261,22 +261,36 @@ def _atomic_save(self, checkpoint, filepath: str):
                 This points to the file that the checkpoint will be stored in.
         """
         tmp_path = str(filepath) + ".part"
-        torch.save(checkpoint, tmp_path)
-        os.replace(tmp_path, filepath)
+        if self.use_tpu:
+            xm.save(checkpoint, tmp_path, master_only=True, global_master=True)
```

Comment:
Maybe add a barrier before `xm.save`, to make sure all processes are in sync?

Reply:
the barrier is already inside `xm.save` [here](https://github.com/pytorch/xla/blob/master/torch_xla/core/xla_model.py#L814)

Reply:
Excuse my naiveness, but shouldn't there be one before save, to make sure that the weights have been updated by every process?